tests/acceptance/pageObjects/FilesPageElement/sharingDialog.js

```diff
@@ -83,10 +83,41 @@ module.exports = {
         .waitForElementVisible('@sharingAutoCompleteDropDownElements')
         .api.elements('css selector', this.elements['sharingAutoCompleteDropDownElements'].selector, (result) => {
           result.value.forEach((value) => {
-            webElementIdList.push(value.ELEMENT)
+            webElementIdList.push(value['ELEMENT'])
           })
         })
         .then(() => webElementIdList)
+    },
+    /**
+     *
+     * @returns {Promise.<string[]>} Array of autocomplete webElementIds
+     */
+    deleteShareWithUserGroup: function (item) {
+      const util = require('util')
+      const deleteSelector = util.format(this.elements.deleteShareButton.selector, item)
+      return this
+        .useXpath()
+        .waitForElementVisible(deleteSelector)
+        .api.element('xpath', deleteSelector, result => {
```

Comment:
why not using the callback of `waitForElementVisible` itself?

Reply:
Using the callback, I dont get the webElementId of the delete button so I couldnot do that or am I doing something wrong

Reply:
why do you need the `webElementId` ? you can use the `this.click()` function with the `deleteSelector`